### ISSUE TITLE
check-synth doesn't flush output

### ISSUE THREAD

**Issue by daniel-larraz** [cvc5/cvc5#9257]
Starting sometime after CVC4 1.7, the `check-synth` command no longer causes output to be flushed. Here is Python code demonstrating the issue:
```
import subprocess
#p = subprocess.Popen(['cvc4-1.7-x86_64-linux-opt', '--lang', 'sygus'], stdout=subprocess.PIPE, stdin=subprocess.PIPE)
#p = subprocess.Popen(['cvc4-1.8-x86_64-linux-opt', '--lang', 'sygus1'], stdout=subprocess.PIPE, stdin=subprocess.PIPE)
p = subprocess.Popen(['cvc5-Linux-2022-11-02-7fb9043', '--lang=sygus'], stdout=subprocess.PIPE, stdin=subprocess.PIPE)
p.stdin.write(b'(set-logic LIA)\n')
p.stdin.write(b'(synth-fun c () Int)\n')
p.stdin.write(b'(check-synth)\n')
p.stdin.flush()
print(p.stdout.readline())
```

With `cvc4-1.7-x86_64-linux-opt` the output is:
```
unsat

```

With `cvc5-Linux-2022-11-02-7fb9043`, there is no output and the test program hangs.

**Comment by ajreynol** [comment]
Related issue: https://github.com/cvc5/cvc5/issues/2720

**Comment by daniel-larraz** [comment]
Like in issue #2720 the problem seems to be parsing the input, not printing the output. I tried the following test suggested in fix #2721 for issue #2720: adding a couple of spaces at the end of the last command, like this:
`p.stdin.write(b'(check-synth  )\n')`.
After this change, the script does not hang anymore.

**Comment by ajreynol** [comment]
After investigating, it appears that the sygus parser hangs in interactive mode due to conditional lexing.  The new flex parser will not have this issue.

Until the flex parser is merged, the current workaround is to use `--std-input-per-line`, or to modify the input like above.
